### ISSUE TITLE
Production tests are failing but say they’re passing

### ISSUE THREAD

**Issue by chasenlehara** [canjs/canjs#4789]
Two tests are failing in the production build, but the test script exits successfully.

Here’s an example in Travis: https://travis-ci.org/canjs/canjs/jobs/511606626

Test output:

```
  5251 passing (43s)
  2 failing

  1) QUnit "test/production.html" on Firefox 65.0.0 / Mac OS X 10.14.0:
       CanJS production tests
         can-event-queue/map
           can listen to DOM events
             stopPropagation works:
     Error: Expected true but was false
  documentClickHandler@http://localhost:3621/dist/bundles/can/test/test.js:62418:17
  dispatch@http://localhost:3621/dist/bundles/can/test/test.js:13611:27
  anonymous/</<@http://localhost:3621/dist/bundles/can/test/test.js:62837:13
  runTest@http://localhost:3621/dist/qunitjs@1.23.1#qunit/qunit:845:14
  run@http://localhost:3621/dist/qunitjs@1.23.1#qunit/qunit:830:4
  run/<@http://localhost:3621/dist/qunitjs@1.23.1#qunit/qunit:972:6
  process@http://localhost:3621/dist/qunitjs@1.23.1#qunit/qunit:631:4
  begin@http://localhost:3621/dist/qunitjs@1.23.1#qunit/qunit:613:2
  resumeProcessing/<@http://localhost:3621/dist/qunitjs@1.23.1#qunit/qunit:673:4
  

  2) QUnit "test/production.html" on Firefox 65.0.0 / Mac OS X 10.14.0:
       CanJS production tests
         can-event-queue/map
           can listen to DOM events
             stopImmediatePropagation works:
     Error: Expected true but was false
  documentClickHandler@http://localhost:3621/dist/bundles/can/test/test.js:62418:17
  dispatch@http://localhost:3621/dist/bundles/can/test/test.js:13611:27
  anonymous/</<@http://localhost:3621/dist/bundles/can/test/test.js:62837:13
  runTest@http://localhost:3621/dist/qunitjs@1.23.1#qunit/qunit:845:14
  run@http://localhost:3621/dist/qunitjs@1.23.1#qunit/qunit:830:4
  run/<@http://localhost:3621/dist/qunitjs@1.23.1#qunit/qunit:972:6
  process@http://localhost:3621/dist/qunitjs@1.23.1#qunit/qunit:631:4
  begin@http://localhost:3621/dist/qunitjs@1.23.1#qunit/qunit:613:2
  resumeProcessing/<@http://localhost:3621/dist/qunitjs@1.23.1#qunit/qunit:673:4
```

**Comment by chasenlehara** [comment]
I looked into this issue today. Some findings:

[This can-dom-events PR](https://github.com/canjs/can-dom-events/pull/64/files) (cc @phillipskevin) added a test that sets up a listener that should never be called: https://github.com/canjs/can-dom-events/blob/395453e566c8f592656ee8212fb9173dd5b9bb85/can-dom-events-test.js#L264-L270

I think the test should move the `removeDelegateListener` and `removeEventListener` calls to right before [done is called](https://github.com/canjs/can-dom-events/blob/395453e566c8f592656ee8212fb9173dd5b9bb85/can-dom-events-test.js#L273). I think this will make sure the listeners get removed regardless of whether the test passes or fails.

Because the `documentClickHandler` was still listening to clicks, this test in `can-event-queue` was causing the handler in `can-dom-events` to be called: https://github.com/canjs/can-event-queue/blob/6582249c97dc38ec35610a6d289737728bdf4c7a/map/map-test.js#L118-L125

This makes `assert.ok(false)` run in the `can-dom-events` tests. The `can-event-queue` test only expects 1 assertion. I would expect an error to be thrown with too many assertions, but I think the failed assertion is associated with the old test, so it doesn’t affect the running test. Since the old test has already been marked as passing, it’s not retroactively “failed” as far as QUnit is concerned. [I’m not 100% sure, that’s how things appear though.]

The result is that errors are reported to Testee but QUnit doesn’t mark any tests as failed, so Testee exits with code 0 and the tests “pass.”

I _think_ Testee should not only check for all the tests to pass, but also check that no other errors were reported. I intend to open a PR that adds this check.

While debugging this, I found that Testee does not appear to have the correct test results either; it has a `tests` array full of objects with the failed test info, but none of the other passing tests. I’m going to verify that and fix it if it’s an issue.

**Comment by phillipskevin** [comment]
Sorry about that @chasenlehara. I published a fix for those event listeners: https://github.com/canjs/can-dom-events/releases/tag/v1.3.9. Let me know if you're still seeing issues.

**Comment by chasenlehara** [comment]
While implementing a change in Testee that would make it exit with a failure code if any failures were reported by testee-client, I realized QUnit 2 doesn’t have this issue. I thought I had already verified the behavior between QUnit 1 & 2 was the same, but apparently I did not test it correctly.

We could go ahead with the change I was planning on implementing, but for now I think the best thing to do is move on from this issue and upgrade to QUnit 2. I filed an issue for adding QUnit 2 support to steal-qunit: https://github.com/stealjs/steal-qunit/issues/31
